[bitnami/elasticsearch] do not fail if asked to chown read-only files

ianroberts opened the pull request:
<!--
 Before you open the request please review the following guidelines and tips to help it be more easily integrated:

 - Describe the scope of your change - i.e. what the change does.
 - Describe any known limitations with your change.
 - Please run any tests or examples that can exercise your modified code.

 Thank you for contributing! We will try to test and integrate the change as soon as we can, but be aware we have many GitHub repositories to manage and can't immediately respond to every request. There is no need to bump or check in on a pull request (it will clutter the discussion of the request).

 Also don't be worried if the request is closed or not integrated sometimes the priorities of Bitnami might not match the priorities of the pull request. Don't fret, the open source community thrives on forks and GitHub makes it easy to keep your changes in a forked repo.
 -->

### Description of the change

Add `|| true` to `chown` commands that operate recursively on folders that are expected to be mounted into the container from volumes or bind mounts, so that the `chown` does not cause a fatal error if any of the target directories or their children are mounted read-only.

### Benefits

Currently when running as root the elasticsearch container will fail to start if any folder under `/opt/bitnami/elasticsearch/config` is mounted from a read-only filesystem.  In particular this happens in the corresponding bitnami Helm chart, where the TLS certificates for Elasticsearch are mounted in from a Kubernetes secret.  This change means that such cases are no longer fatal errors - the files in the read-only filesystems will not of course have their ownership changed, but there's not really anything better we can do if the filesystem is read-only.

### Possible drawbacks

If a mounted filesystem should have been read-write but was mounted read-only by mistake, it would previously have caused a fatal error, prompting the user to change their configuration; this change will make that a no-op so they might not notice.

### Applicable issues

- fixes #77525

### Additional information

This fix should probably be applied before https://github.com/bitnami/charts/pull/31960 is merged.

ianroberts committed:
[bitnami/elasticsearch] do not fail if asked to chown read-only files

Added "|| true" to chown commands so they do not cause the container startup to fail if any of the target folders or their subdirectories are mounted from a read-only filesystem

Signed-off-by: Ian Roberts <i.roberts@sheffield.ac.uk>

ianroberts commented:
Still relevant.

migruiz4 reviewed (commented):
Thank you very much for your contribution @ianroberts. I'm sorry for the late response.

I would like to suggest two changes to your contribution:

- You used `|| :`, but we prefer using `|| true`.
- You added both `|| :` and `-f` option for chown, which is redundant. The option `-f` suppresses both error exit-code and messages, while the `|| :` would supress the exit code. I would suggest not using `-f`, since hiding error message could make troubleshooting really complicated in some cases.

ianroberts commented:
Very sensible suggestions, I've implemented them and rebased on the latest main branch.

ianroberts commented:
> * You added both `|| :` and `-f` option for chown, which is redundant. The option `-f` suppresses both error exit-code and messages, while the `|| :` would supress the exit code.

My first attempt just used the `-f` but it turns out that `-f` suppresses the _messages_ but does not on its own affect the exit code. So I added the `|| :` (now `|| true`) as well to actually fix the issue.

ianroberts commented:
Still waiting for final review.

ianroberts commented:
Not stale, still waiting for final approval by @migruiz4

ianroberts commented:
Not stale. I addressed the review comments six weeks ago on 11th March but my fixes are still awaiting final approval. Please could someone look at this soon as I have another PR on the helm chart that is blocked waiting for this one.

ianroberts commented:
> Not stale. I addressed the review comments six weeks ago on 11th March but my fixes are still awaiting final approval. Please could someone look at this soon as I have another PR on the helm chart that is blocked waiting for this one.

Eight weeks.

@carrodher if @migruiz4 is no longer available to handle this, would you be able to assign someone else?

carrodher commented:
Sorry for the delay, this PR was somehow out of our radar. Could you please move the changes to the `elasticsearch/9/debian-12` folder, which is the one available in Bitnami at this moment? Thanks

ianroberts commented:
Not stale - rebased as requested, awaiting final review.

ianroberts commented:
Rebased a month ago as requested, still waiting for approval and merge.

docker-creator commented:
Hope this gets priority soon, and [#31960](https://github.com/bitnami/charts/pull/31960) as well. Not having auto reload on certificate renewal is a critical problem.

migruiz4 reviewed (approved):
I'm very sorry for the late response. Looks good to me.

ianroberts commented:
Do you know when an image including this fix is likely to be available on Docker hub?  I need to update https://github.com/bitnami/charts/pull/31960 with the fixed image tag before that one can be merged.

migruiz4 commented:
The fix was included in the latest released image `bitnami/elasticsearch:9.0.3-debian-12-r1`
